apps/prairielearn/assets/scripts/question.ts

```diff
@@ -164,3 +170,64 @@ function updateStatus(submission) {
   }
   display.innerHTML = label;
 }
+
+function setupDynamicObjects() {
+  // Install on page load and reinstall on websocket re-render
+  document.querySelectorAll('a.disable-on-click').forEach((link) => {
+    link.addEventListener('click', () => {
+      link.classList.add('disabled');
+    });
+  });
+  // Enable popover
+  $('[data-toggle="popover"]').popover({ sanitize: false, container: 'body' });
+
+  try {
+    const countdownData = decodeData<{
+      serverTimeLimitMS: number;
+      serverRemainingMS: number;
+    }>('submission-suspended-data');
+    new Countdown(
+      '#submission-suspended-display',
+      '#submission-suspended-progress',
+      countdownData.serverRemainingMS,
+      countdownData.serverTimeLimitMS,
+      null, // No Ajax update
+      () => {
+        (document.querySelector('.question-grade') as HTMLButtonElement).disabled = false;
+        (
+          document.querySelectorAll(
+            '.submission-suspended-msg, .grade-rate-limit-popover'
+          ) as NodeListOf<HTMLElement>
+        ).forEach((elem) => {
+          elem.style.display = 'none';
+        });
+      }
+    );
+  } catch (err) {
+    // If there is no submission-suspended-data object, return
+  }
```

Comment:
I don't love using exceptions for control flow here. Consider using a check for a `#submission-suspended-data` element instead.

Reply:
Done in 29dd854102ee537c520e69742fece2c97a021d48.